### DISCUSSION TITLE
[@types/trusted-types] 2.0.3 breaking changes from 2.0.2?

### DISCUSSION THREAD

**Discussion by jeremymeng** [DefinitelyTyped/DefinitelyTyped#65444] [Issues with a @types package]
After upgrading from 2.0.2, we got a build error for the `createHTML` callback.  

>src/xml.browser.ts(22,18): error TS7006: Parameter 's' implicitly has an 'any' type.

https://github.com/Azure/azure-sdk-for-js/blob/b2e7b22a7a9ae4aa3fe664788df7086387be8c8b/sdk/core/core-xml/src/xml.browser.ts#L19-L24

more details at https://github.com/Azure/azure-sdk-for-js/issues/24872

[a comment by an automated account is not shown]

**Comment by bjarkler** [comment]
I'm not sure what change in the type definitions could have caused this, but it seems to me like the compiler is now surfacing a legitimate issue in your code that may have been previously silenced for some reason.

The correct fix would be to change
```ts
 createHTML: (s) => s,
```
to
```ts
 createHTML: (s: string) => s,
```
in [azure-sdk-for-js](https://github.com/Azure/azure-sdk-for-js/blob/b2e7b22a7a9ae4aa3fe664788df7086387be8c8b/sdk/core/core-xml/src/xml.browser.ts#L19-L24).

**Reply by jeremymeng** [reply]
> but it seems to me like the compiler is now surfacing a legitimate issue in your code that may have been previously silenced for some reason.

Agree that it's a legitimate issue. However, the only change we have is upgrading from 2.0.2 to 2.0.3.  Downgrading back to 2.0.2 makes the error going away.
